Chat/assignment-chat/server-chat/src/main/java/ru/assignment/net/ServerClass.java

```diff
@@ -0,0 +1,101 @@
+package ru.assignment.net;
+
+import ru.assignment.model.ChatModel;
+
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.net.ServerSocket;
+import java.net.Socket;
+import java.util.HashSet;
+import java.util.Iterator;
+import java.util.Set;
+
+/**
+ * Created by Андрей on 18.02.2015.
+ */
+public class ServerClass implements Runnable {
+    private final int port;
+    private ChatModel chatModel;
+    private ServerSocket serverSocket;
+    private Set<Session> sessionSet;
+    private int sessionCount;
+
+    public ServerClass(int port, ChatModel chatModel) {
+        this.port = port;
+        this.chatModel = chatModel;
+        sessionSet = new HashSet<Session>();
+    }
+
+    public void run() {
+        try {
+            serverSocket = new ServerSocket(port);
+            try {
+                System.out.println("beforeServerClassrun");
+                while (!serverSocket.isClosed()) {
+                    System.out.println("beforeServerClassaccept");
+                    Socket socket = serverSocket.accept();
+                    System.out.println("afterServerClassaccept completed");
+                    startNewSession(socket, chatModel);
+                }
+                System.out.println("before close connection after while run() ServerClass");
+                checkConnections();
+            } catch (IOException e) {
```

Comment:
Please use timeout check instead of doing everything in IOException

ServerSocket serverSocket = new ServerSocket(port);
serverSocket.setSoTimeout(SO_TIMEOUT);

while (!stopRequested) {
    try {
        Socket socket = serverSocket.accept();
        // ...
    } catch (SocketTimeoutException e) {
        // accept timed out, continue listening unless stop requested
    }